libcontainer/process_linux.go

```diff
@@ -229,13 +231,58 @@ func (p *initProcess) start() (err error) {
 	if err := p.sendConfig(); err != nil {
 		return newSystemError(err)
 	}
-	// wait for the child process to fully complete and receive an error message
-	// if one was encoutered
-	var ierr *genericError
-	if err := json.NewDecoder(p.parentPipe).Decode(&ierr); err != nil && err != io.EOF {
+
+	var (
+		procSync syncType
+		sentRun  bool
+		ierr     *genericError
+	)
+
+loop:
+	for {
+		if err := json.NewDecoder(p.parentPipe).Decode(&procSync); err != nil {
+			if err == io.EOF {
+				break loop
+			}
+			return newSystemError(err)
+		}
+
+		switch procSync {
+		case procStart:
+			break loop
+		case procReady:
+			if err := p.manager.SetLate(); err != nil {
```

Comment:
The issue with doing sets late is that the error for the cgroup not working properly is delayed until the process is almost done setting up. Which results in us having to fake the `.Set()` errors in `.Apply()`. TBH, I never understood why the controllers always called `.Set()` in `.Apply()`. I'll take a look at this later. I have a feeling some of the CPU cgroups will be a bit iffy.